### ISSUE TITLE
Gulp task using CSS injection triggers a full page reload after running another task that uses .reload()

### ISSUE THREAD

**Issue by troywarr** [BrowserSync/browser-sync#535]
I recently refactored my gulpfile:

https://gist.github.com/troywarr/199908fd29c8c47d269e

to take better advantage of parallel task running. It seems to work as intended, except for one issue: After calling `browserSync.reload` via `gulp.watch` for a task that doesn't use CSS injection, the latter no longer works.

In other words, I can run `gulp default` at the command line to start everything up, make any number of changes to my LESS files and see CSS injection work as intended each time, but as soon as I change a script, Jade file, etc. (which triggers `browserSync.reload` via `gulp.watch` on those files), every subsequent change to a LESS file will then cause a full page reload instead of (or at least in addition to) using CSS injection.

I've read and re-read the docs to try to be sure that I've set up BrowserSync correctly. FWIW, the actual changes I made recently didn't touch the `'serve'` task, but were mainly to end my reliance on [gulp-run-sequence](https://github.com/OverZealous/run-sequence) and instead use task dependencies to manage run order. Also let me know if it's useful to see the previous version of my gulpfile, which doesn't exhibit this issue.

**Comment by shakyShane** [comment]
The recommended way to reload following tasks is either to wrap `.reload` in a separate task, or for files that don't require processing, simply use `.on`

```js
gulp.task('bs-reload', function () {
    browserSync.reload();
});


gulp.task('watch:icons', function() {
    return gulp.watch(paths.src + "/icons/**/*", ['jade-html', 'jade-js', 'bs-reload']);
});

// or

gulp.task('watch:ng-html', function() {
    return gulp.watch(paths.src + "/jade/templates/html/**/*.jade", ['ng-html']).on('change', function(){
        browserSync.reload();
    });
});

```

all that being said, the next release also comes with a `.stream()` for separating the two concerns which will help further. :)

**Comment by troywarr** [comment]
Thanks, @shakyShane! I appreciate the quick & helpful reply.

I replaced all instances of `browserSync.reload` with a reference to a new `'bs-reload'` task, and that seems to have fixed the issue.

A couple of concerns, though:

* The [current docs](http://www.browsersync.io/docs/gulp/#gulp-reload) do seem to recommend the approach I'd taken - referencing `browserSync.reload` directly in the `gulp.watch` task array. Do those docs just need to be updated? If so, I'd be happy to submit a PR this weekend.
* My understanding is that the task array passed to `gulp.watch` runs in parallel, not in series. In the case of a long-running task, it seems that that could lead to a race condition where BrowserSync reloads before the other tasks are completed, meaning the reloaded page wouldn't contain all of the latest changes. Without resorting to [`gulp-run-sequence`](https://github.com/OverZealous/run-sequence) or waiting for Gulp 4 (which will provide additional methods to make it easier to manage task order), is there a way to mitigate that?

**Comment by troywarr** [comment]
In fact, as I've been working on my project, my second concern has been validated - the `'bs-reload'` task usually finishes first, leaving the other tasks to complete after the browser has already been reloaded. Will open a separate ticket for that.
